### PR TITLE
Add CFGProcessor and CFGVisualizeLauncher.

### LINKED ISSUE #2354: JavaSource2CFGDOT not generating the DOT representation of the CFG of a given method
I'm using JavaSource2CFGDOT.java to create the DOT representation of the control flow graph of a given method.

I've cloned the project from GitHub, executed  `./gradlew cloneAndBuildDependencies` and `./gradlew assemble`.  I opened the project on IntelliJ, configured `JavaSource2CFGDOT.java` as the main class, and used `path_to_desktop/Test.java path_to_desktop/ -method test -class Test -pdf` as program arguments. 

Test.java is the follow:
```
class Test {
    void test(boolean b1) {
        int x = 0;
        if (b1) {
            x = 1;
        } else {
            x = 2;
        }
    }
}

```

If i try to run JavaSource2CFGDOT with the arguments above, I get the following output:
```
/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/bin/java "-javaagent:/Applications/IntelliJ IDEA.app/Contents/lib/idea_rt.jar=65487:/Applications/IntelliJ IDEA.app/Contents/bin" -Dfile.encoding=UTF-8 -classpath /Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/charsets.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/deploy.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/ext/cldrdata.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/ext/dnsns.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/ext/jaccess.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/ext/jfxrt.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/ext/localedata.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/ext/nashorn.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/ext/sunec.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/ext/sunjce_provider.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/ext/sunpkcs11.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/ext/zipfs.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/javaws.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/jce.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/jfr.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/jfxswt.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/jsse.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/management-agent.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/plugin.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/resources.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/jre/lib/rt.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/lib/ant-javafx.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/lib/dt.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/lib/javafx-mx.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/lib/jconsole.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/lib/packager.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/lib/sa-jdi.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home/lib/tools.jar:/Users/bruno/Desktop/checker-framework/dataflow/out/production/classes:/Users/bruno/Desktop/checker-framework/javacutil/out/production/classes:/Users/bruno/Desktop/checker-framework/checker-qual/out/production/classes:/Users/bruno/.gradle/caches/modules-2/files-2.1/org.checkerframework/compiler/2.4.0/7e0d09e36c241636cc656755199a6e1c665701ed/compiler-2.4.0.jar org.checkerframework.dataflow.cfg.JavaSource2CFGDOT /Users/bruno/Desktop/Test.java /Users/bruno/Desktop/ -method test -class Test -pdf
warning: No SupportedSourceVersion annotation found on org.checkerframework.dataflow.cfg.JavaSource2CFGDOT$1, returning RELEASE_6.
warning: Supported source version 'RELEASE_6' from annotation processor 'org.checkerframework.dataflow.cfg.JavaSource2CFGDOT$1' less than -source '1.8'
warning: No SupportedAnnotationTypes annotation found on org.checkerframework.dataflow.cfg.JavaSource2CFGDOT$1, returning an empty set.
3 warnings
ERROR: Method not found.
Working on Test.java...

Process finished with exit code 1
```

I've noticed that `getMethodTreeAndCompilationUnit(inputFile, method, clas)` in method `generateDOTofCFG` returns null. Therefore, parameters `m.getKey()` and `m.getValue()` in `generateDOTofCFG()` will also be null, resulting in `ERROR: Method not found.`

### PR DESCRIPTION
**Pull request by xingweitian**
Replace `JavaSource2CFGDOT.java` with `CFGVisualizeLauncher.java`. Add `CFGProcessor.java`. 
Fix #2354.
Contributors: @CharlesZ-Chen @xingweitian.

### CODE REVIEW

File: dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGProcessor.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import com.sun.source.util.TreePathScanner;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import org.checkerframework.javacutil.BasicTypeProcessor;
+import org.checkerframework.javacutil.TreeUtils;
+
+/**
+ * The CFG processor running during compilation to generate the control flow graph of a given method
```
Comment (xingweitian) [inline dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGProcessor.java]: This processor runs during...

File: dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGProcessor.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    /** Class Tree. */
+    private ClassTree classTree;
+    /** Method Tree. */
+    private MethodTree methodTree;
+
+    /** Result of CFG process */
+    private CFGProcessResult result;
+
+    /**
+     * Class constructor.
```
Comment (xingweitian) [inline dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGProcessor.java]: Create a CFG processor.

**Comment by xingweitian** [comment]
> I pulled in typetools/master, so I do not understand why GitHub is showing 128 commits...
> 
> @xingweitian Can you add a reference to the `CFGVisualizeLauncher` to the dataflow manual and maybe also in the CF manual in the writing your own checker section?

I pulled again, now everything is good.

File: docs/manual/creating-a-checker.tex [hunk 3]
```diff
@@ -1848,6 +1848,24 @@
 
 \end{itemize}
 
+\noindent
+You can also use \refclass{dataflow/cfg}{CFGVisualizeLauncher} to generate a DOT
+or String representation of the control flow graph of a given method in a given class:
+
+\begin{smaller}
+\begin{Verbatim}
+java -classpath "$CHECKERFRAMEWORK/checker/dist/checker.jar:" \
```
Comment (smillst) [inline docs/manual/creating-a-checker.tex]: This command does not work:
1. The classpath needs to be on the same line.
2. With Java 8, I get the following error:
```
 Exception in thread "main" java.lang.NoSuchFieldError: ANNOTATION_PROCESSOR_MODULE_PATH
	at com.sun.tools.javac.file.Locations.initHandlers(Locations.java:1976)
	at com.sun.tools.javac.file.Locations.<init>(Locations.java:145)
	at com.sun.tools.javac.file.BaseFileManager.createLocations(BaseFileManager.java:115)
	at com.sun.tools.javac.file.BaseFileManager.<init>(BaseFileManager.java:76)
```
3. With Java11, I get the following error:
```
error: Could not find class file for 'LambdaWrapped'.
1 error
ERROR: internal error in type processor! method typeProcessOver() doesn't get called.
```

Comment (xingweitian) [reply docs/manual/creating-a-checker.tex]: @smillst Thanks for your review! That command indeed does not work with Java 8, I will improve it.
With Java 11, it looks like the class `LambdaWrapped` does not exist in the target Java file.

Comment (smillst) [reply docs/manual/creating-a-checker.tex]: > With Java 11, it looks like the class LambdaWrapped does not exist in the target Java file.

Yes, that was the problem.  I was able to get the following command to work with Java 11:
```
java -cp $CHECKERFRAMEWORK/checker/dist/checker.jar org.checkerframework.dataflow.cfg.CFGVisualizeLauncher   LambaWrapped.java output/ -class LambaWrapped -method getStreamOfWrappedFunctions -pdf
```

File: dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGVisualizeLauncher.java [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+ * use it.
+ */
+public class CFGVisualizeLauncher {
+
+    /**
+     * The main entry point of CFGVisualizeLauncher.
+     *
+     * @param args the passed arguments
+     */
+    public static void main(String[] args) {
```
Comment (smillst) [inline dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGVisualizeLauncher.java]: There isn't any documentation here on what args can be passed.  Maybe just add a reference to the `printUsage` method here?

**Pull request merged**